src/core/lib/security/credentials/external/external_account_credentials.cc

```diff
@@ -167,7 +167,16 @@ void ExternalAccountCredentials::OnExchangeTokenInternal(grpc_error* error) {
   } else {
     if (options_.service_account_impersonation_url.empty()) {
       metadata_req_->response = ctx_->response;
```

Comment:
The whole response was set so the content is actually correct and we've already check it.
This fix is to have a deep copy of `hdrs` and we need to check if the address has a single reference, which seems hard to do in unit tests.

Reply:
btw, I'll add a comprehensive integration test once the implementation finalized, and this kind of memory issues can be easily catched.